### PR TITLE
Fill data for placeholder species

### LINKED ISSUE #2291: Add let's go eevee and let's go pikachu species
I think they fit and should be added to the pokémon expansion.
They should differ from regular ones on:
Species ID
Not being able to breed
Different movesets/learnset
Base stats
Shiny locked (not sure if this is out of scope)
Unable to evolve

Additionally, this could be an excuse to make something like not being able to deposit a certain species ID in a storage box, but I feel like that should be handled in a different github issue.

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
The final ~~(I promise!)~~ follow up PR to #4230.

- [x] Adds data for:
  - [x] Extra Mothim forms.
    - [x] They evolve from and have offspring based off Burmy's form.
  - [x] Extra Scatterbug/Spewpa forms.
    - [x] They evolve from and have offspring based off Vivillon's form
  - [x] Totem Pokémon
    - [x] Added `isTotem` flag, to generate 3 perfect IVs.
    - [x] Single-stage totems had to have their offspring hardcoded to avoid getting more Totems via breeding.
    - [x] Alolan Raticate
    - [x] Alolan Marowak
    - [x] Gumshoos
    - [x] Vikavolt
    - [x] Ribombee
    - [x] Araquanid
    - [x] Lurantis
    - [x] Salazzle
    - [x] Togedemaru
    - [x] Mimikyu
      - [x] Added missing Busted form.
    - [x] Kommo-O
  - [x] Partner Pikachu/Eevee (fixes #2291)
- [x] Made all species IDs absolute instead of relative, to avoid confusion when adding new species.
- [x] Moved shared dex text to its own file.
- [x] Renamed `species_info/gen_X.h` to `species_info/gen_X_families.h`
- [x] Added missing entries for Hidden Abilities that default to `ABILITY_NONE`.

## Feature(s) this PR does NOT handle:
Insert data for Shadow Lugia. This should come in #4128.

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

File: test/battle/move_effect/rage_fist.c [hunk 1]
```diff
@@ -252,6 +252,10 @@ SINGLE_BATTLE_TEST("Rage Fist base power is not increased if move had no affect"
 SINGLE_BATTLE_TEST("Rage Fist base power is increased if Disguise breaks")
 {
     s16 timesGotHit[2];
+    u16 species = SPECIES_NONE;
+
+    PARAMETRIZE { species = SPECIES_MIMIKYU_DISGUISED; }
+    PARAMETRIZE { species = SPECIES_MIMIKYU_TOTEM_DISGUISED; }
 
     GIVEN {
         PLAYER(SPECIES_MIMIKYU_DISGUISED) { Ability(ABILITY_DISGUISE); }
```
Comment (AlexOn1ine) [inline test/battle/move_effect/rage_fist.c:261]: Use `species` var for player.

File: src/data/pokemon/species_info.h [hunk 2]
```diff
@@ -1,181 +1,5 @@
...
-    "This form is the most aggressive,\n"
-    "bombarding enemies with the\n"
-    "intensity of flames blazing within a hearth.");
-
-const u8 gOgerponCornerstoneMaskPokedexText[] = _(
-    "In this form, it draws on the power\n"
-    "of stone. Its body is rock-solid,\n"
-    "protecting it from all manner of\n"
-    "attacks.");
+#include "species_info/shared_dex_text.h"
```
Comment (AlexOn1ine) [inline src/data/pokemon/species_info.h:2]: any reason for having them in a new file?

Comment (AsparagusEduardo) [reply src/data/pokemon/species_info.h:2]: Mainly that it's getting awkward to scroll through a growing list of shared entries to reach either `SPECIES_NONE`, `SPECIES_EGG` or the includes to the families.

File: test/battle/move_effect/rage_fist.c (hunk 1 above)
Comment (AsparagusEduardo) [reply test/battle/move_effect/rage_fist.c:261]: Whoops

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**